src/RapideFix/Validation/FixVersionValidator.cs

```diff
@@ -0,0 +1,36 @@
+﻿using RapideFix.DataTypes;
+using System;
+
+namespace RapideFix.Validation
+{
+  public class FixVersionValidator : IValidator
+  {
+    public bool IsValid(Span<byte> data, FixMessageContext messageContext)
+    {
+      if(!data.StartsWith(KnownFixTags.FixVersion))
+      {
+        return false;
+      }
+      int beginStringLength = data.IndexOf(Constants.SOHByte);
+      var fixVersionValue = data.Slice(KnownFixTags.FixVersion.Length, beginStringLength - KnownFixTags.FixVersion.Length);
```

Comment:
Let's consider `"{1}={2}|"`, where `{1}` represents the tag and `{2}` the value.
In this case `beginStringLength` would be the length of `"{1}={2}"`, so `beginStringLength - KnownFixTags.FixVersion.Length` will be the length of `"={2}"`, so you are slicing a one character longer than needed.
Am I right?

Reply:
That is a tricky, because the FixVersion's definition includes a | SOH char upfront which it should not. As I am adding unit tests, this something I am fixing.